integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/ViewPopRedirectTests/RedirectTest.java

```diff
@@ -104,17 +104,26 @@ public void testRedirect() throws Exception {
         requestHeader.put("Accept", "application/json");
 
         HttpRequestUtil.doPost(new URL(getProxyServiceURLHttp(PROXY_NAME)), inputPayload, requestHeader);
+
+        //Check if the processor has deactivated
         Awaitility.await()
                 .pollInterval(3, TimeUnit.SECONDS)
                 .atMost(120, TimeUnit.SECONDS)
                 .until(isProcessorDeactivated(PROCESSOR_NAME));
         Assert.assertFalse(messageProcessorClient.isActive(PROCESSOR_NAME), "Message processor" + PROCESSOR_NAME
                 + "should not be active, but it is active.");
+
+        //Manually deactivate REDIRECT_PROCESSOR and check if deactivated properly
```

Comment:
do we need this line messageProcessorClient.deactivateProcessor(REDIRECT_PROCESSOR_NAME);

after Assert.assertFalse(messageProcessorClient.isActive(PROCESSOR_NAME), "Message processor" + PROCESSOR_NAME
                + "should not be active, but it is active."); since we assert that it is deactivated

Reply:
This is an entirely different Message processor that we are deactivating. It is required that we deactivate it manually for the rest of the test to work

Reply:
This is an entirely different Message processor that we are deactivating. It is required that we deactivate it manually for the rest of the test to work.